### PR TITLE
Log preferred hosts

### PR DESCRIPTION
**Pull request by mbautin**
Log preferred host names in case when running a task on a non-preferred one, because this sometimes indicates an inconsistency between the hostnames Spark workers and Hadoop datanodes use.

### CODE REVIEW

File: core/src/main/scala/spark/scheduler/cluster/TaskSetManager.scala [hunk 1]
```diff
@@ -201,7 +201,9 @@ private[spark] class TaskSetManager(
           val taskId = sched.newTaskId()
           // Figure out whether this should count as a preferred launch
           val preferred = isPreferredLocation(task, host)
-          val prefStr = if (preferred) "preferred" else "non-preferred"
+          val prefStr = if (preferred) "preferred" else
```
Comment (tomdz) [inline core/src/main/scala/spark/scheduler/cluster/TaskSetManager.scala]: Scala preferred style is to wrap the two lines in `{}` for the else branch, or combine them into one line and move the else in front of it.

Comment (mateiz) [inline core/src/main/scala/spark/scheduler/cluster/TaskSetManager.scala]: Agreed with Tom on this. Otherwise it looks good.

**Comment by mbautin** [comment]
This should be ready to merge.

**Comment by mateiz** [comment]
Great, thanks!

**Comment by mateiz** [comment]
By the way, the style you used was still not quite standard; here's what Tom meant for the right style for multi-line conditionals:

```
      val prefStr = if (preferred) {
        "preferred"
      } else {
        "non-preferred, not one of " + task.preferredLocations.mkString(", ")
      }
```

I will change it to this style myself.

**Pull request merged**